nucypher/utilities/prometheus/collector.py

```diff
@@ -309,9 +311,22 @@ def initialize(self, metrics_prefix: str, registry: CollectorRegistry) -> None:
             self.metrics[metric_key] = metric_class(metric_name, metric_doc, registry=registry)
 
     def _collect_internal(self) -> None:
-        events = self.event_filter.get_new_entries()
-        for event in events:
-            self._event_occurred(event)
+        from_block = self.filter_last_block_checked
+        to_block = self.contract_agent.blockchain.client.block_number
+        if from_block == to_block:
+            # nothing to see here
+            return
+
+        events_throttler = ContractEventsThrottler(agent=self.contract_agent,
+                                                   event_name=self.event_name,
+                                                   from_block=from_block,
+                                                   to_block=to_block,
+                                                   **self.filter_arguments)
+        for event_record in events_throttler:
+            self._event_occurred(event_record.raw_event)
+
+        # update last block for the next round
+        self.filter_last_block_checked = to_block
```

Comment:
If the `to_block` for the query range is indeed inclusive (see @fjarri 's comment - https://github.com/nucypher/nucypher/pull/2322/files#r497823819)- this makes me think that this line would need to change to be:

```
self.filter_last_block_checked = to_block + 1
```

to ensure that the events for the same block isn't repeated during the next iteration. If so, other changes will need to be made to accommodate for this. Interested to get thoughts here?

Reply:
Thanks for pointing this out @fjarri! 😅 

The range is indeed inclusive, which may lead to duplicate events being obtained. It has now been fixed.

Here is an easy CLI test I ran locally to confirm:
```
derek @ Dereks-MacBook-Pro-2 ~/Documents/Github/repos/forks/derek/nucypher
 [20] → nucypher status events --network ibex --provider https://rinkeby.infura.io/v3/1e738a40d1b24d01b6bc7fb10fd9af3c --contract-name StakingEscrow --event-name CommitmentMade
Reading Latest Chaindata...
Showing events from block 7290043 to latest

--------- StakingEscrow Events ---------

CommitmentMade:
...
15000000000000000000000, block_number: 7292850

```

```
(nucypher) (base)
derek @ Dereks-MacBook-Pro-2 ~/Documents/Github/repos/forks/derek/nucypher
 [21] → nucypher status events --network ibex --provider https://rinkeby.infura.io/v3/1e738a40d1b24d01b6bc7fb10fd9af3c --contract-name StakingEscrow --event-name CommitmentMade --from-block 7292849 --to-block 7292850
Reading Latest Chaindata...
Showing events from block 7292849 to 7292850

--------- StakingEscrow Events ---------

CommitmentMade:
  - (EventRecord) staker: 0x488998c6dc3bA8d0Aee0A1952290d751C85eD730, period: 18537, value: 15000000000000000000000, block_number: 7292850
```

```
(nucypher) (base)
derek @ Dereks-MacBook-Pro-2 ~/Documents/Github/repos/forks/derek/nucypher
 [22] → nucypher status events --network ibex --provider https://rinkeby.infura.io/v3/1e738a40d1b24d01b6bc7fb10fd9af3c --contract-name StakingEscrow --event-name CommitmentMade --from-block 7292850 --to-block 7292851
Reading Latest Chaindata...
Showing events from block 7292850 to 7292851

--------- StakingEscrow Events ---------

CommitmentMade:
  - (EventRecord) staker: 0x488998c6dc3bA8d0Aee0A1952290d751C85eD730, period: 18537, value: 15000000000000000000000, block_number: 7292850
```

```
(nucypher) (base)
derek @ Dereks-MacBook-Pro-2 ~/Documents/Github/repos/forks/derek/nucypher
 [26] → nucypher status events --network ibex --provider https://rinkeby.infura.io/v3/1e738a40d1b24d01b6bc7fb10fd9af3c --contract-name StakingEscrow --event-name CommitmentMade --from-block 7292850 --to-block 7292850
Reading Latest Chaindata...
Showing events from block 7292850 to 7292850

--------- StakingEscrow Events ---------

CommitmentMade:
  - (EventRecord) staker: 0x488998c6dc3bA8d0Aee0A1952290d751C85eD730, period: 18537, value: 15000000000000000000000, block_number: 7292850
```

Reply:
✔️